src/instructlab/sdg/llmblock.py

```diff
@@ -113,21 +132,30 @@ def generate(self, samples, **gen_kwargs) -> Dataset:
         # validate each sample
         for sample in samples:
             if not self._validate(self.prompt_template, sample):
-                return None
+                logger.warning("Sample failed validation")  #TODO add details
+                #TODO remove sample from samples
+
+        if len(samples) == 0:
+            return Dataset.from_list([])
 
         # generate the output
-        outputs = []
-        if batched:
-            outputs = self._generate(samples, **gen_kwargs)
-        else:
-            outputs = [self._generate([sample], **gen_kwargs)[0] for sample in samples]
-        logger.debug("Generated outputs: {}".format(outputs))
+
+        outputs = self._generate(samples, **gen_kwargs)
+        logger.debug("Generated outputs: %s", outputs)
+
+        num_parallel_samples = gen_kwargs.get("n", 1)
```

Comment:
Hmm, and here's a reason to make num_parallel_samples part of the block config ... and add 'n' to gen_kwargs based on that